ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -2742,51 +2782,79 @@ private RelNode genTableLogicalPlan(String tableAlias, QB qb) throws SemanticExc
+          RelOptHiveTable optTable = new RelOptHiveTable(relOptSchema, fullyQualifiedTabName,
+              rowType, tabMetaData, nonPartitionColumns, partitionColumns, virtualCols, conf,
+              partitionCache, colStatsCache, noColsMissingStats);
+          
+          final HiveTableScan hts = new HiveTableScan(cluster,
+              cluster.traitSetOf(HiveRelNode.CONVENTION), optTable,
+              null == tableAlias ? tabMetaData.getTableName() : tableAlias,
+                getAliasId(tableAlias, qb),
+                HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_CBO_RETPATH_HIVEOP),//TODOY
+                qb.isInsideView() || qb.getAliasInsideView().contains(tableAlias.toLowerCase()));
+          
+          if (tableType == TableType.DRUID) {
+            // Build Druid query
+            String address =
+                HiveConf.getVar(conf, HiveConf.ConfVars.HIVE_DRUID_BROKER_DEFAULT_ADDRESS);
+            String dataSource = tabMetaData.getParameters().get(Constants.DRUID_DATA_SOURCE);
+            Set<String> metrics = new HashSet<>();
           RexBuilder rexBuilder = cluster.getRexBuilder();
           RelDataTypeFactory dtFactory = rexBuilder.getTypeFactory();
-          List<RelDataType> druidColTypes = new ArrayList<>();
-          List<String> druidColNames = new ArrayList<>();
+            List<RelDataType> druidColTypes = new ArrayList<>();
+            List<String> druidColNames = new ArrayList<>();
           //@TODO FIX this, we actually do not need this anymore,
           // in addition to that Druid allow numeric dimensions now so this check is not accurate
-          for (RelDataTypeField field : rowType.getFieldList()) {
+            for (RelDataTypeField field : rowType.getFieldList()) {
             if (DruidTable.DEFAULT_TIMESTAMP_COLUMN.equals(field.getName())) {
               // Druid's time column is always not null.
               druidColTypes.add(dtFactory.createTypeWithNullability(field.getType(), false));
             } else {
               druidColTypes.add(field.getType());
             }
-            druidColNames.add(field.getName());
-            if (field.getName().equals(DruidTable.DEFAULT_TIMESTAMP_COLUMN)) {
-              // timestamp
-              continue;
-            }
-            if (field.getType().getSqlTypeName() == SqlTypeName.VARCHAR) {
-              // dimension
-              continue;
+              druidColNames.add(field.getName());
+              if (field.getName().equals(DruidTable.DEFAULT_TIMESTAMP_COLUMN)) {
+                // timestamp
+                continue;
+              }
+              if (field.getType().getSqlTypeName() == SqlTypeName.VARCHAR) {
+                // dimension
+                continue;
+              }
+              metrics.add(field.getName());
             }
-            metrics.add(field.getName());
-          }
-
+            // TODO: Default interval will be an Interval once Calcite 1.15.0 is released.
```

Comment:
Not part of this patch, but since we are moving it, can we remove this TODO? It does not seem relevant anymore.